internal/support/notifications/rest_notification.go

```diff
@@ -45,24 +45,38 @@ func notificationHandler(w http.ResponseWriter, r *http.Request) {
 
 	LoggingClient.Info("Posting Notification: " + n.String())
 	n.Status = models.NotificationsStatus(models.New)
-	n.ID, err = dbClient.AddNotification(n)
+
+	op := notification.NewAddExecutor(dbClient, n)
+	n.ID, err = op.Execute()
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusConflict)
+		switch err.(type) {
+		case errors.ErrNotificationInUse:
+			http.Error(w, err.Error(), http.StatusConflict)
+		default:
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
 		LoggingClient.Error(err.Error())
 		return
 	}
 
 	if n.Severity == models.NotificationsSeverity(models.Critical) {
```

Comment:
I think I would advocate including this logic in the AddExecutor. The executor should encapsulate all of the relevant logic for the handler, and this is the only place where we check the severity and send right away. Also note that this is the only place the Distributor is used, so you may be able to just make that functionality a private function on the `notificationAdd` type. This would eliminate the newly added global var, interface and `New` constructor.

Reply:
Agree, much more elegant solution. I will edit this commit